tools/crowtty/src/main.rs

```diff
@@ -12,40 +13,83 @@ pub struct Chunk {
     buf: Vec<u8>,
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    // let mut dport = None;
-
-    // for port in serialport::available_ports().unwrap() {
-    //     if let serialport::SerialPortType::UsbPort(serialport::UsbPortInfo {
-    //         serial_number: Some(sn),
-    //         ..
-    //     }) = &port.port_type
-    //     {
-    //         if sn.as_str() == "ajm001" {
-    //             dport = Some(port.clone());
-    //             break;
-    //         }
-    //     }
-    // }
-
-    // let dport = if let Some(port) = dport {
-    //     port
-    // } else {
-    //     eprintln!("Error: No `Pellegrino` connected!");
-    //     return Ok(());
-    // };
-
-    // let mut port = serialport::new(dport.port_name, 115200)
-    //     .timeout(Duration::from_millis(5))
-    //     .open()
-    //     .map_err(|_| "Error: failed to create port")?;
+enum Connect {
+    Serial(Box<dyn SerialPort>),
+    Tcp(TcpStream),
+}
 
-    let mut carry = Vec::new();
+impl std::io::Write for Connect {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        match self {
+            Connect::Serial(s) => s.write(buf),
+            Connect::Tcp(t) => t.write(buf),
+        }
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        match self {
+            Connect::Serial(s) => s.flush(),
+            Connect::Tcp(t) => t.flush(),
+        }
+    }
+}
+
+impl std::io::Read for Connect {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        match self {
+            Connect::Serial(s) => s.read(buf),
+            Connect::Tcp(t) => t.read(buf),
+        }
+    }
+}
 
-    let mut port = TcpStream::connect("127.0.0.1:9999").unwrap();
+impl Connect {
+    fn new_from_tcp(port: u16) -> Self {
+        let port = TcpStream::connect(&format!("127.0.0.1:{port}")).unwrap();
+        port.set_read_timeout(Some(Duration::from_millis(10))).ok();
 
-    // port.set_timeout(Duration::from_millis(10)).ok();
-    port.set_read_timeout(Some(Duration::from_millis(10))).ok();
+        Connect::Tcp(port)
+    }
+
+    fn new_from_serial(path: &str, baud: u32) -> Self {
+        let port = serialport::new(path, baud)
+            .timeout(Duration::from_millis(10))
+            .open()
+            .unwrap();
+        Connect::Serial(port)
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = std::env::args().skip(1).collect::<Vec<_>>();
+    let args = args.iter().map(String::as_str).collect::<Vec<_>>();
+
+    let mut port = match args.as_slice() {
```

Comment:
nit: at some point it might be nice to just use `clap` for arg parsing here --- we already use it in `melpomene`. not a blocker for this PR though...

Reply:
Agreed! Happy to swap to that in the future esp if we have more options.